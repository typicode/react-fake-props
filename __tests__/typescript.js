import path from 'path'
import fakeProps from '../src'

const ComponentFile = path.resolve(
  './fixtures/typescript/Component.tsx'
)
const SimpleFile = path.resolve('./fixtures/typescript/Simple.tsx')
const NoProps = path.resolve('./fixtures/typescript/NoProps.tsx')
const Multiple = path.resolve('./fixtures/typescript/Multiple.tsx')

describe('fakeProps', () => {
  it('should return an object with no props (snapshot)', () => {
    expect(fakeProps(NoProps)).toMatchSnapshot()
  })

  it('should return an object with required props faked (snapshot)', () => {
    expect(fakeProps(ComponentFile)).toMatchSnapshot()
  })

  it('should return an object with all props faked (snapshot)', () => {
    expect(fakeProps(ComponentFile, { optional: true })).toMatchSnapshot()
  })

  it('should return a person on calling getFriend prop function (test function generation)', () => {
    const allProps = fakeProps(ComponentFile, { optional: true })
    const friend = allProps.getFriend()

    expect(friend.name).toBeDefined()
    expect(friend.name).toBe('getFriend.name')
    expect(friend.age).toBe(1)
  })

  it('should return more props with optional', () => {
    const allProps = fakeProps(ComponentFile, { optional: true })
    const requiredProps = fakeProps(ComponentFile)

    const allPropsLength = Object.keys(allProps).length
    const requiredPropsLength = Object.keys(requiredProps).length

    expect(allPropsLength).toBeGreaterThan(requiredPropsLength)
  })

  it('should return an object with shape props faked', () => {
    const props = fakeProps(SimpleFile)
    const { requiredString, requiredObject } = props

    expect(props).toEqual({
      requiredString,
      requiredObject: {
        requiredString: requiredObject.requiredString
      }
    })
  })

  it('should prefix shape props', () => {
    const props = fakeProps(SimpleFile)
    expect(props.requiredObject.requiredString).toBe(
      'requiredObject.requiredString'
    )
  })

  it('should work with multiple components', () => {
    const component = fakeProps(Multiple, { all: true }).find(
      ({ displayName }) => displayName === 'First'
    )
    expect(component.props.requiredString).toBe('requiredString')
  })
})
