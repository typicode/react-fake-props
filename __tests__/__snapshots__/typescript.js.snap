// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`fakeProps should return an object with all props faked (snapshot) 1`] = `
Object {
  "getFriend": [Function],
  "onNameChange": [Function],
  "person": Object {
    "address": Object {
      "pincode": 1,
      "street": "person.address.street",
      "street2": "person.address.street2",
    },
    "age": 1,
    "family": Array [
      1,
    ],
    "id": 1,
    "name": "person.name",
    "repos": Array [
      Object {
        "access": "public",
        "commits": Array [
          Object {
            "id": "person.repos.commits.id",
            "message": "person.repos.commits.message",
          },
        ],
        "demoSite": "person.repos.demoSite",
        "npm": "person.repos.npm",
        "stars": 0,
        "url": "person.repos.url",
      },
    ],
  },
}
`;

exports[`fakeProps should return an object with no props (snapshot) 1`] = `Object {}`;

exports[`fakeProps should return an object with required props faked (snapshot) 1`] = `
Object {
  "onNameChange": [Function],
  "person": Object {
    "address": Object {
      "pincode": 1,
      "street": "person.address.street",
    },
    "age": 1,
    "id": 1,
    "name": "person.name",
    "repos": Array [
      Object {
        "access": "public",
        "commits": Array [
          Object {
            "id": "person.repos.commits.id",
            "message": "person.repos.commits.message",
          },
        ],
        "stars": 0,
        "url": "person.repos.url",
      },
    ],
  },
}
`;
